assets/js/modules/adsense/settings/settings-edit.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * AdSense Settings Edit component.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { useEffect } from '@wordpress/element';
+import { addFilter, removeFilter } from '@wordpress/hooks';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { STORE_NAME } from '../datastore/constants';
+import SettingsForm from './settings-form';
+import ProgressBar from '../../../components/progress-bar';
+const { useSelect, useDispatch } = Data;
+
+export default function SettingsEdit() {
+	const canSubmitChanges = useSelect( ( select ) => select( STORE_NAME ).canSubmitChanges() );
+	const isDoingSubmitChanges = useSelect( ( select ) => select( STORE_NAME ).isDoingSubmitChanges() );
+	const haveSettingsChanged = useSelect( ( select ) => select( STORE_NAME ).haveSettingsChanged() );
+
+	// Rollback any temporary selections to saved values on dismount.
```

Comment:
Is this the older rollback solution we were using—eg. didn't @aaemnnosttv update this in Analytics for an approach that involved a component higher-up in the tree? (I think it's now in `AnalyticsMain`, and would go in `SettingsMain` here.)

EDIT: Yup, see: https://github.com/google/site-kit-wp/blob/develop/assets/js/modules/analytics/settings/settings-main.js#L34-L42